### ISSUE TITLE
Babel browser.js warning

### ISSUE THREAD

**Issue by edwinvdgraaf** [rolaveric/karma-systemjs#22]
Starting this morning karma couldn't resolve babel anymore. Resulting that my tests don't run anymore.

It starts with this error;  WARN [watcher]: Pattern "/node_modules/babel/node_modules/babel-core/browser.js" does not match any file.

I'm guessing that it has something to do with [this line](https://github.com/rolaveric/karma-systemjs/blob/master/lib/index.js#L124). From the path in the warning I expect that babel had a dependency on the babel-core? However I cannot find a commit which introduced this change.

Or did I do something wrong? Let me know if I can be of any assistance.

**Comment by rolaveric** [comment]
What's happening is karma-systemjs looks up what your transpiler is and adds it to the list of files that Karma serves for testing.  It will try to find it based on the the `paths.[transpilerName]` in your SystemJS config, or will revert to using `require.resolve()` and a known relative path (which is the `/node_modules/babel-core/browser.js` in that line).
That warning states that karma was told to serve files that match that pattern, but it can't find any.

I just did a fresh `npm install` on my [angular-seed fork](https://github.com/rolaveric/angular-seed/tree/es6) and checked that nothing was outdated (`npm outdated --depth 0`).  Couldn't replicate your issue.

What concerns me is that path - it's an absolute path from the file system root.
eg. Here's the same error when coerced from my checkout:
WARN [watcher]: Pattern "/home/rolaveric/WebstormProjects/angular-seed/app/bower_components/angular/meep.js" does not match any file.
But I tried all sorts of combinations with the `baseURL` and `paths.babel` within the SystemJS config and couldn't replicate it.

Any chance you can provide:
- The `baseURL` from your SystemJS config
- The `paths.babel` from your SystemJS config
- The rest of the error messages (I assume there are more)

**Comment by edwinvdgraaf** [comment]
Thanks for your elaborate answer.  First of, I somewhat made the warning line shorter by cleaning up the path. Which was stupid of me, sorry for causing confusing but the complete warning I'm getting is: 

```
WARN [watcher]: Pattern "/Users/edwin/Projects/html/mcud-website/node_modules/babel/node_modules/babel-core/browser.js" does not match any file.
```

And the error of karma in total is:
```
INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
WARN [watcher]: Pattern "/Users/edwin/Projects/html/mcud-website/node_modules/babel/node_modules/babel-core/browser.js" does not match any file.
INFO [PhantomJS 1.9.8 (Mac OS X)]: Connected on socket 4Qgw3wFC2j-M_UHR5MT4 with id 12789697
WARN [web-server]: 404: /base/babel.js
WARN [web-server]: 404: /base/babel.js
PhantomJS 1.9.8 (Mac OS X) ERROR
  Error loading "test/spec/Calendar.spec" at /Users/edwin/Projects/html/mcud-website/test/spec/Calendar.spec.js
  Error loading "babel" at /Users/edwin/Projects/html/mcud-website/babel.js
  Not Found: /Users/edwin/Projects/html/mcud-website/babel.js

Finished in 0.096 secs / 0 secs

SUMMARY:
✔ 0 tests completed
```


Thanks for providing a test project ([angular-seed-fork](https://github.com/rolaveric/angular-seed/tree/es6)), which I've also ran and indeed this works correctly. So now that I have a base to compare I have  `npm` installs babel-core correct for [angular-seed-fork](https://github.com/rolaveric/angular-seed/tree/es6) and runs the test correctly. However it doesn't turn up in my own project.

```
(angular-seed-fork) $ ls node_modules/babel/node_modules
babel-core chokidar commander convert-source-map fs-readdir-recursive glob lodash output-file-sync path-is-absolute slash source-map
```

While the yields of `ls` in my own project is:
```
(mcud-website) $ ls node_modules/babel/node_modules
chokidar commander convert-source-map fs-readdir-recursive output-file-sync path-is-absolute slash source-map
```

Debuggin a bit further I noticed that my package.json lists both babel-core and babel itself.

```
  "devDependencies": {
    "babel": "^5.2.17",
    "babel-core": "^5.3.1",
```

Removing babel-core solved my issue.

Bottom line: I'd never assumed this would make npm skip the installation of babel-core for the babel dependency. I think this issue can be closed, thanks again for your explanation and effort on this plugin. :)

**Comment by rolaveric** [comment]
Hold on, just one thing though: Do you have a `paths.babel` property in your SystemJS config?
That should get around any issues with whether you have babel or babel-core installed.

Also going by what you've described, I should change karma-systemjs to lookup `require.resolve('babel-core') + '/browser.js'`, rather than expecting it to be included as a dependency of 'babel'.
After all, that's where the file we want is located now.

**Comment by edwinvdgraaf** [comment]
The paths.babel solution is what I'm currently using because I need babel-core for systemjs-builder.

The proposed change would help me, however requires all karma-systemjs users to update their package.json I suppose?

**Comment by rolaveric** [comment]
That's true, but I think it's necessary.  And it should encourage people to set paths.babel.
I'll bump the version number up to 0.4.0, update the README, and make sure the error message is helpful.
